x/rewards/keeper/providers.go

```diff
@@ -121,3 +124,110 @@ func (k Keeper) specProvidersBasePay(ctx sdk.Context, chainID string) ([]types.B
 	}
 	return basepays, totalBasePay
 }
+
+// ContributeToValidatorsAndCommunityPool transfers some of the providers' rewards to the validators and community pool
+// the function return the updated reward after the participation deduction
+func (k Keeper) ContributeToValidatorsAndCommunityPool(ctx sdk.Context, reward math.Int, senderModule string) (updatedReward math.Int, err error) {
+	// calculate validators and community participation fractions
+	validatorsParticipation, communityParticipation, err := k.CalculateContributionPercentages(ctx, reward, senderModule)
+	if err != nil {
+		return reward, err
+	}
+
+	if communityParticipation.Equal(sdk.OneDec()) {
+		err := k.FundCommunityPoolFromModule(ctx, reward, senderModule)
+		if err != nil {
+			return reward, utils.LavaFormatError("failed funding the community pool with whole reward", err,
+				utils.Attribute{Key: "reward", Value: reward.String()},
+				utils.Attribute{Key: "community_participation", Value: communityParticipation.String()},
+			)
+		}
+		return sdk.ZeroInt(), nil
+	}
+
+	// send validators participation
+	validatorsParticipationReward := validatorsParticipation.MulInt(reward).TruncateInt()
+	if !validatorsParticipationReward.IsZero() {
+		coins := sdk.NewCoins(sdk.NewCoin(k.stakingKeeper.BondDenom(ctx), validatorsParticipationReward))
+		err = k.bankKeeper.SendCoinsFromModuleToModule(ctx, senderModule, k.feeCollectorName, coins)
+		if err != nil {
+			return reward, utils.LavaFormatError("sending validators participation failed", err,
+				utils.Attribute{Key: "validators_participation_reward", Value: coins.String()},
+				utils.Attribute{Key: "validators_participation", Value: validatorsParticipation.String()},
+				utils.Attribute{Key: "reward", Value: reward.String()},
+			)
+		}
+	}
+
+	// send community participation
+	communityParticipationReward := communityParticipation.MulInt(reward).TruncateInt()
+	if !communityParticipationReward.IsZero() {
+		coins := sdk.NewCoins(sdk.NewCoin(k.stakingKeeper.BondDenom(ctx), communityParticipationReward))
+		err = k.bankKeeper.SendCoinsFromModuleToModule(ctx, senderModule, k.feeCollectorName, coins)
```

Comment:
you send to the same address
also unitest to see the community pool got what it deserves

Reply:
done. see https://github.com/lavanet/lava/commit/94c283788ff3e853f7089d16f8c8de229ba045d2